### DISCUSSION TITLE
Seeking Assistance for Implementing Object Transformation Features in C++ With IfcOpenShell

### DISCUSSION THREAD

**Discussion by jahyun-koo** [IfcOpenShell/IfcOpenShell#4079] [Q&A]
I am currently aiming to develop a utility feature that manipulates IFC files directly using IfcOpenShell.

I am interested in adjusting the dimensions and scale of specific objects through C++ code. My goal is to convert the relevant code to WebAssembly (wasm) and execute this on a web platform.

In my research, I noticed that the BlenderBIM tool seems to offer this functionality through its transform feature, being able to adjust scale and dimensions. However, it appears that this feature is implemented in Python.

I am seeking advice on identifying the corresponding C++ code. I have attempted to peruse the BlenderBIM source code, however, it is rather difficult to ascertain exactly which code corresponds to this feature.

I would greatly appreciate any advice or guidance you could provide.

**Comment by aothms** [comment]
I think you have two options: either use C++ and compile to WASM, but also consider this https://github.com/IfcOpenShell/wasm-preview/ which allows you to use the full extent of all python code from within JS, which might really be beneficial if you consider using the high level editing functionality. Over time we'll likely find ways to make the pyodide approach lighter.

Most of the high level python code doesn't have a direct equivalent in C++, but would require reimplementing it using the low-level entity instance manipulation that the C++ ifcopenshell library offers.

**Reply by jahyun-koo** [reply]
Thank you for your response. I have greatly benefited from the help of the IfcOpenShell project. Given your suggestion, I will refer to the wasm-preview project and implement it accordingly.
